clients/src/main/java/org/apache/kafka/clients/admin/RemoveMembersFromConsumerGroupResult.java

```diff
@@ -0,0 +1,95 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.clients.admin;
+
+import org.apache.kafka.common.KafkaFuture;
+import org.apache.kafka.common.internals.KafkaFutureImpl;
+import org.apache.kafka.common.message.LeaveGroupRequestData.MemberIdentity;
+import org.apache.kafka.common.protocol.Errors;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * The result of the {@link Admin#removeMembersFromConsumerGroup(String, RemoveMembersFromConsumerGroupOptions)} call.
+ *
+ * The API of this class is evolving, see {@link Admin} for details.
+ */
+public class RemoveMembersFromConsumerGroupResult {
+
+    KafkaFuture<Map<MemberIdentity, Errors>> future;
+    private List<MemberIdentity> memberIdentities;
+
+    RemoveMembersFromConsumerGroupResult(KafkaFuture<Map<MemberIdentity, Errors>> future,
+                                         List<MemberIdentity> memberIdentities) {
+        this.future = future;
+        this.memberIdentities = memberIdentities;
+    }
+
+    /**
+     * Returns a future which indicates whether the request was 100% success, i.e. no
+     * either top level or member level error.
+     */
+    public KafkaFuture<Void> all() {
+        final KafkaFutureImpl<Void> result = new KafkaFutureImpl<>();
+        this.future.whenComplete((memberErrors, throwable) -> {
+            if (throwable != null) {
+                result.completeExceptionally(throwable);
+            } else {
+                for (MemberIdentity memberToRemove : memberIdentities) {
+                    Throwable exception = getSubLevelError(memberErrors, memberToRemove);
+                    if (exception != null) {
+                        result.completeExceptionally(exception);
+                        return;
+                    }
+                }
+                result.complete(null);
+            }
+        });
+        return result;
+    }
+
+    /**
+     * Returns the selected member future.
+     */
+    public KafkaFuture<Void> memberResult(MemberIdentity member) {
+        final KafkaFutureImpl<Void> result = new KafkaFutureImpl<>();
+        this.future.whenComplete((memberErrors, throwable) -> {
+            if (throwable != null) {
+                result.completeExceptionally(throwable);
+            } else {
+                Throwable exception = getSubLevelError(memberErrors, member);
+                if (exception != null) {
+                    result.completeExceptionally(exception);
+                } else {
+                    result.complete(null);
+                }
+            }
+        });
+        return result;
+    }
+
+    static <K> Throwable getSubLevelError(Map<K, Errors> subLevelErrors, K subKey) {
+        if (!subLevelErrors.containsKey(subKey)) {
```

Comment:
The error message was not very clear since it is too general; in previous commits we have more explicit error messages like "member / partition were not included in the returned response, indicating it was not accepted or processed at the broker side".

Reply:
I'd suggest we just make one static function at the KafkaAdminClient level, and then at each Result interpret the `IllegalArgumentException` with more explicit error message.